features/admin.tenants.v1/components/dropdown/tenant-dropdown.tsx

```diff
@@ -230,7 +246,51 @@ const TenantDropdown: FunctionComponent<TenantDropdownInterface> = (props: Tenan
             setTempTenantAssociationsList(association.associatedTenants);
         }
         setTenantAssociations(association);
-    }, [ associatedTenants, defaultTenant, tenantDomain ]);
+    }, [ associatedTenants, defaultTenant, currentTenant ]);
+
+    const {
+        data: deploymentUnitResponse,
+        isLoading: isDeploymentUnitsLoading,
+        error: deploymentUnitFetchRequestError
+    } = useDeploymentUnits(
+        {
+            revalidateIfStale: true
+        },
+        isCentralDeploymentEnabled
+    );
+
+    useEffect(() => {
+        setDeploymentUnits(deploymentUnitResponse?.deploymentUnits);
```

Comment:
Let's directly return deploymentUnits from the `useDeploymentUnits` hook so that we can get rid of this useEffect

Reply:
As discussed offline, agreed to follow the existing approach